jdaviz/core/marks.py

```diff
@@ -233,23 +347,82 @@ def __init__(self, shadowing, shadow_width=1, **kwargs):
                          colors=[kwargs.pop('color', 'white')],
                          **kwargs)
 
-        # sync initial values
-        for attr in self._sync_traits:
-            setattr(self, attr, getattr(shadowing, attr))
-
-        # keep values synced when traits on shadowing object change
-        shadowing.observe(self._on_shadowing_changed)
+        self._setup_shadowing(shadowing,
+                              ['scales', 'x', 'y', 'visible', 'line_style', 'marker'],
+                              ['stroke_width', 'marker_size'])
 
     def _on_shadowing_changed(self, change):
-        attr = change['name']
-        if attr[0] in ['stroke_width', 'marker_size']:
-            value = change['new'] + self._shadow_width if change['new'] else 0
-        elif attr not in self._sync_traits:
+        super()._on_shadowing_changed(change)
+        if change['name'] in ['stroke_width', 'marker_size']:
+            # apply the same, but increased by the shadow width
+            setattr(self, change['name'],
+                    change['new'] + self._shadow_width if change['new'] else 0)
+        return
+
+
+class ShadowLabelFixedY(Label, ShadowMixin):
+    """
+    Label whose position shadows that of a parent ``shadowing`` line and will flip alignment
+    based on whether it is left or right of the center of the viewer.
+    """
+    def __init__(self, viewer, shadowing, shadow_traits=['visible'],
+                 y=0.95, point_index=0, **kwargs):
+        super().__init__(**kwargs)
+        self._viewer = viewer
+        self.y = [y]
+        self.scales['y'] = LinearScale(min=0, max=1)
+        self._point_index = point_index
+        self._setup_shadowing(shadowing,
+                              shadow_traits,
+                              ['x', 'scales', 'labels', 'colors'])
+
+        viewer.state.add_callback("x_min", lambda x_min: self._update_align())
+        viewer.state.add_callback("x_max", lambda x_max: self._update_align())
+
+    def _force_redraw(self):
+        # TODO: bug in bqplot that change in align/colors traitlet doesn't update immediately,
+        # we'll get around it in the meantime by just forcing the Label to see a change to the
+        # text traitlet
+        text = self.text
+        self.text = ['']
+        self.text = text
```

Comment:
not clear why this isn't working.  It seems that the styling is at least supposed to be listening to changes to both color and align: https://github.com/bqplot/bqplot/blob/master/js/src/Label.ts#L24

Reply:
Should I withhold review until this mystery is solved? We're probably doing funky stuff in our own viewers overriding the default behavior somehow?

Reply:
I don't think we need to wait. If/when we resolve this, the only change in this PR would be removing this method (and all calls to it). But if you or anyone considers this work-around unmergeable, then we can dig deeper into the underlying cause now.

Reply:
Let's just get this in as-is, but if you have another follow-up issue on this somewhere, please cross-link here if you have not already. Thanks!